app/domain/authentication/common/validate_resource_restrictions.rb

```diff
@@ -0,0 +1,58 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    ValidateResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions: ExtractResourceRestrictions.new,
+        logger:                        Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id host_name account constraints authentication_request)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::ValidatingResourceRestrictions.new(@host_name))
+
+        resource_restrictions = extract_resource_restrictions
+        validate_resource_restrictions_configuration(resource_restrictions)
+        validate_request_matches_resource_restrictions(resource_restrictions)
+
+        @logger.debug(LogMessages::Authentication::ValidatedResourceRestrictions.new)
+      end
+
+      private
+
+      def extract_resource_restrictions
+        @extract_resource_restrictions.call(
+          authenticator_name: @authenticator_name,
+          service_id:         @service_id,
+          host_name:          @host_name,
+          account:            @account
+        )
+      end
+
+      def validate_resource_restrictions_configuration(resource_restrictions)
+        @logger.debug(LogMessages::Authentication::ValidatingResourceRestrictionsConfiguration.new)
+
+        @constraints.validate(
```

Comment:
i think we should add a comment on what we expect `@constraints` to be as it's an input parameter and not a dependency. Future developer will need to explicitly create this object themselves so the documentation on it should be explicit.

Reply:
done